modules/images/fetchpriority/load.php

```diff
@@ -0,0 +1,49 @@
+<?php
+/**
+ * Module Name: Fetchpriority
+ * Description: Adds `fetchpriority=high` parameter to the non lazy loaded images.
+ * Experimental: Yes
+ *
+ * @since n.e.x.t
+ * @package performance-lab
+ */
+
+/**
+ * Filters an image tag in content to add the fetchpriority attribute if it is not lazy-loaded.
+ *
+ * @since n.e.x.t
+ *
+ * @param string $filtered_image The image tag to filter.
+ * @param string $context        The context of the image.
+ * @return string The filtered image tag.
+ */
+function fetchpriority_img_tag_add_attr( $filtered_image, $context ) {
+
+	if ( 'the_content' !== $context && 'the_post_thumbnail' !== $context ) {
+		return $filtered_image;
+	}
+
+	if ( ! empty( $filtered_image ) && strpos( $filtered_image, 'loading="lazy"' ) === false && strpos( $filtered_image, 'fetchpriority=' ) === false ) {
```

Comment:
Since we rely on the presence of `loading="lazy"` here, we first of all have to make sure the site has lazy-loading enabled in general (via `wp_lazy_loading_enabled( 'img', $context )`). Otherwise, this will cause a bug on sites that disabled it, effectively giving every image `fetchpriority="high"` which would be detrimental. Let's add a test to cover that scenario as well.

Reply:
👍🏼 added a check above @ L26